org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -203,6 +206,51 @@ private synchronized void doNodeInitialisation() {
             logger.debug("{}: Using static definition with existing {} channels", nodeIeeeAddress, nodeChannels.size());
         }
 
+        // Add statically defined endpoints and clusters
+        for (Channel channel : nodeChannels) {
+            // Process the channel properties
+            Map<String, String> properties = channel.getProperties();
+            int endpointId = Integer.parseInt(properties.get(ZigBeeBindingConstants.CHANNEL_PROPERTY_ENDPOINT));
+            ZigBeeEndpoint endpoint = node.getEndpoint(endpointId);
+            if (endpoint == null) {
+                int profileId;
+                if (properties.get(ZigBeeBindingConstants.CHANNEL_PROPERTY_PROFILEID) == null) {
+                    profileId = ZigBeeProfileType.ZIGBEE_HOME_AUTOMATION.getKey();
+                } else {
+                    profileId = Integer.parseInt(properties.get(ZigBeeBindingConstants.CHANNEL_PROPERTY_PROFILEID));
+                }
+
+                logger.debug("{}: Creating statically defined device endpoint {} with profile {}", nodeIeeeAddress,
+                        endpointId, ZigBeeProfileType.getByValue(profileId));
+                endpoint = new ZigBeeEndpoint(coordinatorHandler.getNetworkManager(), node, endpointId);
+                endpoint.setProfileId(profileId);
+                node.addEndpoint(endpoint);
+            }
+
+            List<Integer> staticClusters;
+            boolean modified = false;
+            staticClusters = processClusterList(endpoint.getInputClusterIds(),
```

Comment:
I must admit, that it was a little hard for me to understand what is happening here. I think the source of this - for me - is that the intention of `processClusterList` was not evident to me, maybe because the method name is somewhat generic, maybe because the list returned by the method may mean different things (if the list is empty, it signifies that no additional clusters were statically defined, otherwise it contains the list of all statically defined and non-statically defined clusters), maybe something else.

I have toyed around with the code, trying to make it easier to read from my personal view, and came up with the following code (replacing lines 230--246 and the method `processClusterList`) - do you also think that it makes the code somewhat easier to understand? Or is this just personal preference?

```
            boolean modifiedClusterIds = false;

            Collection<Integer> staticInputClusterIds = getStaticClusterIds(properties,
                    ZigBeeBindingConstants.CHANNEL_PROPERTY_INPUTCLUSTERS);
            if (!endpoint.getInputClusterIds().containsAll(staticInputClusterIds)) {
                logger.debug("{}: Forcing endpoint {} input clusters {}", nodeIeeeAddress, endpointId,
                        staticInputClusterIds);
                staticInputClusterIds.addAll(endpoint.getInputClusterIds());
                endpoint.setInputClusterIds(new ArrayList<Integer>(staticInputClusterIds));
                modifiedClusterIds = true;
            }

            Collection<Integer> staticOutputClusterIds = getStaticClusterIds(properties,
                    ZigBeeBindingConstants.CHANNEL_PROPERTY_OUTPUTCLUSTERS);
            if (!endpoint.getOutputClusterIds().containsAll(staticOutputClusterIds)) {
                logger.debug("{}: Forcing endpoint {} output clusters {}", nodeIeeeAddress, endpointId,
                        staticOutputClusterIds);
                staticOutputClusterIds.addAll(endpoint.getOutputClusterIds());
                endpoint.setOutputClusterIds(new ArrayList<Integer>(staticOutputClusterIds));
                modifiedClusterIds = true;
            }
```

where 
```
    private Set<Integer> getStaticClusterIds(Map<String, String> channelProperties, String propertyName) {
        return Arrays.stream(channelProperties.get(propertyName).split(",")).map(Integer::new)
                .collect(Collectors.toSet());
    }
```